utils.c

```diff
@@ -589,7 +589,6 @@ load_file_data (int     fd,
       if (res < 0)
         {
           errsv = errno;
-          close (fd);
           errno = errsv;
           return NULL;
         }
```

Comment:
I think this can collapse down to

```
if (res < 0)
  return NULL;
```

since it's pointless to save `errno` and then immediately restore it. After doing that, the declaration of `errsv` also becomes unused and should be dropped.